### PR TITLE
feat(expr): support streaming `bit_and` and `bit_or` aggregate

### LINKED ISSUE #9281: implement `bit_and` / `bit_or` / `bit_xor` for smallint / int / bigint
```
bit_xor ( smallint ) → smallint
bit_xor ( integer ) → integer
bit_xor ( bigint ) → bigint
```
Computes the bitwise exclusive OR of all non-null input values. 

Can be useful as a checksum for an unordered set of values. This is useful for testing.

### PR DESCRIPTION
**Pull request by wangrunji0408**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

resolve #9281 

This PR adds streaming version `bit_and` and `bit_or` aggregate functions. The state of both functions is a list of 0/1 counts for each bit, so that input values can be retracted.

Limitation: Due to the difference of state and output type, 2-phase aggregation for both functions are disabled.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [x] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

Support the following aggregate functions in materialized views:
```
bit_and ( smallint ) → smallint
bit_and ( integer ) → integer
bit_and ( bigint ) → bigint

bit_or ( smallint ) → smallint
bit_or ( integer ) → integer
bit_or ( bigint ) → bigint
```

### CODE REVIEW

File: src/expr/impl/src/aggregate/bit_and.rs [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+/// drop materialized view mv;
+///
+/// statement ok
+/// drop table t;
+/// ```
+#[derive(Debug, Default, Clone)]
+struct BitAndI32Updatable;
+
+#[aggregate("bit_and(int4) -> int4", state = "int8[]")]
+impl BitAndI32Updatable {
```
Comment (wangrunji0408) [inline src/expr/impl/src/aggregate/bit_and.rs:184]: Seems that we could avoid duplicate code using generic. 🤔

```rust
#[aggregate("bit_and(int2) -> int2", state = "int8[]")]
#[aggregate("bit_and(int4) -> int4", state = "int8[]")]
#[aggregate("bit_and(int8) -> int8", state = "int8[]")]
impl<T> BitAndUpdatable<T> {
```

**Review by lmatz — APPROVED** [review]
thanks, tests LGTM

**Review by stdrc — COMMENTED** [review]
LGTM except for the duplication

**Review by BugenZhao — APPROVED** [review]
REALLY COOL

**Review by stdrc — APPROVED** [review]
LGTM！

**Pull request merged**